packages/medusa/src/loaders/config.ts

```diff
@@ -9,18 +9,29 @@ const errorHandler = isProduction
     }
   : console.log
 
-export default (rootDirectory: string): ConfigModule => {
```

Comment:
question: The idea is to allow all the properties to be either a `string | () => string | () => Promise<string>` is that the idea?

Reply:
yes, that is the format that typeorm accepts, you can either send a string, a function that returns a string, or a Promise that is eventually resolved. with IAM the callback is necessary as the password token expires every 15 minutes.

Reply:
yeah, so in that case my suggestion should be fine